### ISSUE TITLE
wrong calculation of .ceil() and .floor()

### ISSUE THREAD

**Issue by thinkrapido** [nyx-space/hifitime#145]
```rust
    let epoch = Epoch::from_unix_milliseconds(600f64);

    println!("epoch {}", epoch);
    println!("epoch {}", epoch .ceil(3.minutes()));
    println!("epoch {}", epoch .floor(3.minutes()));

    let epoch = Epoch::now().unwrap();//::from_unix_milliseconds(600f64);

    println!("epoch {}", epoch);
    println!("epoch {}", epoch .ceil(3.minutes()));
    println!("epoch {}", epoch .floor(3.minutes()));
```
creates
```
epoch 1970-01-01T00:00:00.599999904 UTC
epoch 1970-01-01T00:03:00 UTC
epoch 1970-01-01T00:00:00 UTC
epoch 2022-10-03T17:44:29.898032665 UTC
epoch 2022-10-03T17:47:23 UTC                         # error
epoch 2022-10-03T17:44:23 UTC                         # error
```

**Comment by ChristopherRabotin** [comment]
Thanks, that does look like a bug.

To be sure, you'd expect the following output, is that right ?

epoch 2022-10-03T17:42:00 UTC

**Comment by thinkrapido** [comment]
yes, it should be the appropriate fraction with seconds to zero

**Comment by ChristopherRabotin** [comment]
The source of the issue is that the `floor` function will floor the duration that is stored in the `epoch`. However, all epochs are stored in TAI so all of the floor and ceil is done in TAI. There's a non-constant offset between TAI and UTC, hence the rounding you're seeing.

I'm implementing a fix now that adds a ceil and floor function with in a given time system.

**Comment by ChristopherRabotin** [comment]
@thinkrapido The bug is fixed in 3.5.0 in the way you wanted it to be fixed.
